CardScan/Classes/CreditCardOcr/MainLoopStateMachine.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  MainLoopStateMachine.swift
+//  CardScan
+//
+//  Created by Sam King on 8/5/20.
+//
+
+import Foundation
+
+public enum MainLoopState {
+    case initial
+    case ocrOnly
+    case cardOnly
+    case ocrAndCard
+    case finished
+}
+
+public protocol MainLoopStateMachine {
+    func loopState() -> MainLoopState
+    func event(prediction: CreditCardOcrPrediction) -> MainLoopState
+}
+
+open class OcrMainLoopStateMachine: MainLoopStateMachine {
+    
+    var state: MainLoopState = .initial
+    var startTimeForCurrentState = Date()
+    var timeOfFirstOcr: Date?
+    
+    public let errorCorrectionDuration = 2.0
```

Comment:
nit: Name this with the units, so it's clear that this is seconds and not milliseconds.

Reply:
Yup, done.